nemo/collections/nlp/models/language_modeling/megatron_t5_model.py

```diff
@@ -18,15 +18,15 @@
 from pytorch_lightning.trainer.trainer import Trainer
 
 from nemo.collections.nlp.data.language_modeling.megatron.dataset_utils import build_train_valid_test_datasets
-from nemo.collections.nlp.models.language_modeling.megatron_lm_encoder_decoder_model import (
-    MegatronLMEncoderDecoderModel,
+from nemo.collections.nlp.models.language_modeling.megatron_lm_adapter_encoder_decoder_model import (
+    MegatronLMAdapterEncoderDecoderModel,
 )
 from nemo.utils import logging
 
 __all__ = ["MegatronT5Model"]
 
 
-class MegatronT5Model(MegatronLMEncoderDecoderModel):
+class MegatronT5Model(MegatronLMAdapterEncoderDecoderModel):
```

Comment:
Any idea how GPT adapters were able to do this without changing the class inheritance?

Reply:
In GPT case, the wrapper class is created on the top module level. In our case, because several models use T5, I moved down the wrapper to the middle module and I had to make a wrapper on MegatronLMEncoderDecoderModel that enables adapters. Any T5 model can change its inheritance from MegatronLMEncoderDecoderModel to MegatronLMAdapterEncoderDecoderModel to enable adapters.